### PR TITLE
Highlight module references equally

### LINKED ISSUE #166: Highlighting of built-in vs. user-defined modules
[vim-elixir](https://github.com/elixir-lang/vim-elixir) highlights Elixir atoms (module names like `List` or `Plug.Conn`). emacs-elixir not highlighting them is the result of a design choice or of a kinda-bug?

Thanks in advance. I'd open a PR with an available solution (in case this is a bug) if only I had **any** experience with Emacs/elisp. I'm a total newbie :(

### PR DESCRIPTION
**Pull request by rranelli**
This PR sets `module` fontlock to be consistent across references. This PR also removes the notion of highlighting built-in modules differently than *user/custom* modules. 

Concerning the risk of wrongly highlighting capitalized words that do not correspond to module references, the picture below sumarizes the changes in highlighting:

screenshot from 2015-03-18 23 37 37

To the left, the current behavior and to the right the proposed ones.

UPDATE:
I fontified the contents of sigils with `string-font-lock-face` just like the regexp does.

The new result is as follows:

screenshot from 2015-03-20 20 12 35

Fixes #166

### CODE REVIEW

**Comment by rranelli** [comment]
@mattdeboard @tonini @whatyouhide

I've fixed the things I've left pending before, I think this is ok to be properly reviewed now.

**Comment by rranelli** [comment]
ping,

Any thoughts on this PR ?

Thanks!

**Comment by mattdeboard** [comment]
Hi @rranelli 

Sorry about the unresponsiveness, I know it's frustrating.

Can you sort out merge conflicts here please?

**Comment by rranelli** [comment]
@mattdeboard No problem at all =).

I will sort the merge conflicts right now.

**Comment by rranelli** [comment]
Done

File: elixir-mode.el [hunk 1]
```diff
@@ -330,6 +305,33 @@ for the Elixir programming language."
     (,(elixir-rx (group sigils))
      1 font-lock-builtin-face)
 
+    ;; Sigil patterns. Elixir has support for eight different sigil delimiters.
```
Comment (mattdeboard) [inline elixir-mode.el]: It looks like you modified this correctly *but* you left [these lines](https://github.com/rranelli/emacs-elixir/blob/highlight-all-modules-equally/elixir-mode.el#L335-L360) in tact. Please remove them.

File: test/elixir-mode-font-test.el [hunk 2]
```diff
@@ -125,7 +141,8 @@ end"
   (elixir-test-with-temp-buffer
       ":oriole
 :andale
-:ms2pid"
+:ms2pid
+:CapitalizedAtom"
```
Comment (mattdeboard) [inline test/elixir-mode-font-test.el]: You didn't add any `should` checks for the newly added line so adding it didn't really add anything of value.

**Comment by mattdeboard** [comment]
Also I just saw that `~s` is also included in the `heredoc` regex pattern. Is this why `~r` and `~s` are highlighted different colors in your screenshot?

**Comment by mattdeboard** [comment]
To answer my own question, yes it is. Super minor issue though, not going to worry about it.

File: test/elixir-mode-font-test.el (hunk 2 above)
Comment (rranelli) [inline test/elixir-mode-font-test.el]: Bad merge conflict resolution on my part. 
I had removed the line `(should (eq (elixir-test-face-at 26) 'elixir-atom-face))` and now I see that was not a good move. 

Just added it again.

File: elixir-mode.el (hunk 1 above)
Comment (rranelli) [inline elixir-mode.el]: I wasnt sure if sigils and regexes should be highlighted differently. 

Should they look the same then ?

**Comment by mattdeboard** [comment]
Fixes #166

**Pull request merged**